src/cephadm/cephadm.py

```diff
@@ -1618,10 +1617,18 @@ def _port_in_use(af: socket.AddressFamily, address: str) -> bool:
             else:
                 raise e
         return False
-    return any(_port_in_use(af, address) for af, address in (
-        (socket.AF_INET, '0.0.0.0'),
-        (socket.AF_INET6, '::')
-    ))
+
+    if port_ips and str(port_num) in port_ips.keys():
+        ip = port_ips[str(port_num)]
+        if is_ipv6(ip):
+            return _port_in_use(socket.AF_INET6, ip)
+        else:
+            return _port_in_use(socket.AF_INET, ip)
+    else:
```

Comment:
Non-blocking nit: the else here is unnecessary because the if block above always returns, this has the additional benefit of making the diff between the old and new code smaller.

Reply:
removed the else